Assets/Engine/Inventory/Editor/ContainerDescriptorEditor.cs

```diff
@@ -65,14 +65,17 @@ public override void OnInspectorGUI()
             HandleOnlyStoreWhenOpen(onlyStoreWhenOpen);       
         }
         else if(containerDescriptor.hasUi)
-        {  
-            // check if the gameObject has a open animation
-            foreach (AnimatorControllerParameter controllerParameter in containerDescriptor.gameObject.GetComponent<Animator>().parameters)
+        {
+            var animator = containerDescriptor.gameObject.GetComponent<Animator>();
+            if (animator != null)
             {
-                if (controllerParameter.name == "Open")
+                foreach (AnimatorControllerParameter controllerParameter in containerDescriptor.gameObject.GetComponent<Animator>().parameters)
                 {
-                    bool openWhenContainerViewed = EditorGUILayout.Toggle(new GUIContent("open when container viewed", "Set if the open animation should run when the container UI is opened"), containerDescriptor.openWhenContainerViewed);
-                    HandleOpenWhenContainerViewed(openWhenContainerViewed);
+                    if (controllerParameter.name == "Open")
```

Comment:
nitpick: can reduce nesting by 1 here by inverting the if statement.

`if (controllerParameter.name != "Open") { continue; }`

Reply:
done